api/app/Models/PoolCandidate.php

```diff
@@ -1229,4 +1233,92 @@ public function scopeProcessNumber(Builder $query, ?string $processNumber): Buil
 
         return $query;
     }
+
+    public function computeFinalDecision()
+    {
+        $this->load(['user']);
+
+        $status = $this->pool_candidate_status;
+        $decision = null;
+
+        if (in_array($status, PoolCandidateStatus::toAssessGroup())) {
+            $assessmentStatus = $this->computed_assessment_status;
+            $overallStatus = null;
+            if (isset($assessmentStatus['overallAssessmentStatus'])) {
+                $overallStatus = $assessmentStatus['overallAssessmentStatus'];
+            }
+
+            $decision = match ($overallStatus) {
+                OverallAssessmentStatus::QUALIFIED->name => FinalDecision::QUALIFIED_PENDING->name,
+                OverallAssessmentStatus::DISQUALIFIED->name => FinalDecision::DISQUALIFIED_PENDING->name,
+                default => FinalDecision::TO_ASSESS->name
+            };
+        } else {
+
+            $decisionMap = [
```

Comment:
Maybe use the `match` statement again, for consistency? But the logic looks right. 👍